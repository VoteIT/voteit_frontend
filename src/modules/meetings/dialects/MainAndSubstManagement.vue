<template>
  <v-tooltip :text="tooltip" location="bottom start">
    <template #activator="{ props }">
      <span :class="{ 'text-warning': !allAssigned }" v-bind="props">
        {{ potentialVotes }}/{{ group.votes }}
      </span>
    </template>
  </v-tooltip>
</template>

<script lang="ts" setup>
import { computed, PropType } from 'vue'

import useActive from '@/modules/active/useActive'
import { GroupMembership, MeetingGroup } from '../types'
import useMeeting from '../useMeeting'

import { isRoleMembership } from './types'

const props = defineProps({
  group: {
    type: Object as PropType<MeetingGroup & { memberships: GroupMembership[] }>,
    required: true
  }
})

const { meetingId } = useMeeting()
const { activeUserIds } = useActive(meetingId)

const roleMemberships = computed(() => props.group.memberships.filter(isRoleMembership))
const activeVoters = computed(() => roleMemberships.value.filter(m => activeUserIds.value.includes(m.user)))
const potentialVotes = computed(() => Math.min(activeVoters.value.length, props.group.votes || 0))
const allAssigned = computed(() => props.group.votes === potentialVotes.value)

const tooltip = computed(() => {
  return allAssigned.value
    ? 'Potentiella röster baseras på aktiva användare med en roll i gruppen.'
    : 'Denna grupp kan inte använda alla sina röster. Det kan bero på att det inte finns tillräckligt många medlemmar med rösträtt, eller att gruppens medlemmar inte är aktiva i mötet.'
})
</script>
