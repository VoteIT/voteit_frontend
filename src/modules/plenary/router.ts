import Plenary from './Plenary.vue'

export default {
  path: '/p/:id/:roomId/:aid',
  name: 'Plenary',
  component: Plenary
}
