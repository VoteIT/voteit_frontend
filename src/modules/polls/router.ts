import { RouterView } from 'vue-router'

import Poll from './PollView.vue'
import Polls from './PollsView.vue'
import StartPoll from './StartPollView.vue'

export default {
  path: 'polls',
  component: RouterView,
  children: [
    {
      path: '',
      name: 'polls',
      component: Polls
    },
    {
      path: ':pid/:pslug',
      name: 'poll',
      component: Poll
    },
    {
      path: 'new',
      name: 'start-poll',
      component: StartPoll
    },
    {
      path: 'new/:aid',
      name: 'start-poll-ai',
      component: StartPoll
    },
    {
      path: ':state',
      name: 'poll-state',
      component: Polls
    }
  ]
}
