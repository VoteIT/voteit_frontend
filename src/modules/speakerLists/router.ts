import SpeakerListsView from './SpeakerListsView.vue'

export default {
  path: 'lists/:system/:aid',
  name: 'speakerLists',
  component: SpeakerListsView
}
