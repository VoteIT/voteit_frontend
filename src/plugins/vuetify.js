import '@mdi/font/css/materialdesignicons.css'
import 'vuetify/lib/styles/main.sass'
import 'vuetify/lib/styles/styles.sass'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/lib/components'
import * as directives from 'vuetify/lib/directives'
import colors from 'vuetify/lib/util/colors'

const light = {
  dark: false,
  variables: {
    'border-color': '223,207,200',
    'border-opacity': 1
  },
  colors: {
    background: '#f8f4f2',
    'app-bar': '#563661',
    'app-bar-divider': '#765681',
    'app-bar-active': '#7976B7',
    'menu-active': '#9C8afa',
    surface: '#ffffff',
    'surface-active': colors.green.lighten5,
    primary: '#3729d8',
    secondary: colors.grey.darken1,
    'secondary-lighten-2': colors.grey.lighten2,
    accent: colors.indigo.darken4,
    error: colors.deepOrange.base,
    info: colors.lightBlue.base,
    'info-lighten-2': colors.lightBlue.lighten2,
    success: '#6ce498',
    'success-lighten-2': '#cbf6db',
    'success-darken-2': '#1da475',
    warning: colors.red.darken2
  }
}

/* Some day
const dark = {
  dark: true,
  variables: {},
  colors: {
    background: '#111',
    surface: '#1e1e1e',
    primary: colors.indigo.darken2,
    secondary: colors.blueGrey.darken2,
    accent: colors.indigo.base,
    error: colors.deepOrange.darken2,
    info: colors.lightBlue.darken2,
    success: colors.green.darken2,
    warning: colors.red.darken4
  }
}
*/

export default createVuetify({
  components,
  directives,
  theme: {
    defaultTheme: 'light',
    themes: {
      light
      // dark
    }
  }
})
