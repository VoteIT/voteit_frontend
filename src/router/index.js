import { createRouter, createWebHistory } from 'vue-router'
import Home from '../views/Home.vue'
import JoinMeeting from '../views/JoinMeeting.vue'
import meeting from './meeting'

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  meeting,
  {
    path: '/join/:id/:slug',
    name: 'JoinMeeting',
    component: JoinMeeting
  }
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
})

export default router
