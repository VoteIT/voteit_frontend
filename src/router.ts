import { createRouter, createWebHistory } from 'vue-router'

import Home from '@/views/Home.vue'
import meeting from '@/modules/meetings/router'
import JoinMeeting from '@/modules/meetings/JoinView.vue'
import plenary from '@/modules/plenary/router'
import speakerFullscreen from '@/modules/speakerLists/fullscreenRouter'
import AppBar from '@/components/AppBar.vue'

const routes = [
  {
    path: '/',
    name: 'Home',
    components: {
      default: Home,
      appBar: AppBar
    }
  },
  meeting,
  plenary,
  speakerFullscreen,
  // Join has url outside meeting, so users don't need meeting roles to visit this view.
  {
    path: '/join/:id/:slug',
    name: 'JoinMeeting',
    components: {
      default: JoinMeeting,
      appBar: AppBar
    }
  }
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
})

export default router
