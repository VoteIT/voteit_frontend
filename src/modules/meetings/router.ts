import AgendaEditView from '../agendas/AgendaEditView.vue'
import AgendaItemView from '../agendas/AgendaItemView.vue'
import polls from '../polls/router'

import ControlPanelView from './ControlPanel.vue'
import ElectoralRegistersView from './electoralRegisters/ElectoralRegistersView.vue'
import JoinMeeting from './JoinView.vue'
import ParticipantsView from './ParticipantsView.vue'
import StartView from './StartView.vue'
import MeetingView from './MeetingView.vue'
import MinutesView from './MinutesView.vue'

export default [
  {
    path: '/m/:id/:slug',
    name: 'MeetingRouterView',
    component: MeetingView,
    children: [
      {
        path: '',
        name: 'meeting',
        component: StartView
      },
      {
        path: 'settings',
        name: 'settings',
        component: ControlPanelView,
        children: [
          {
            path: ':panel',
            name: 'controlPanel',
            component: ControlPanelView
          }
        ]
      },
      {
        path: 'agenda',
        name: 'agendaEdit',
        component: AgendaEditView
      },
      {
        path: 'p',
        name: 'participants',
        component: ParticipantsView,
        children: [
          {
            path: ':tabId',
            name: 'participantsTab',
            component: ParticipantsView
          }
        ]
      },
      {
        path: 'er',
        name: 'electoralRegisters',
        component: ElectoralRegistersView
      },
      {
        path: 'minutes',
        name: 'meetingMinutes',
        component: MinutesView
      },
      polls,
      {
        path: 'a/:aid/:aslug',
        name: 'agendaItem',
        component: AgendaItemView
      }
    ]
  },
  // Join has url outside meeting, so users don't need meeting roles to visit this view.
  {
    path: '/join/:id/:slug',
    name: 'meeting:join',
    component: JoinMeeting
  }
]
