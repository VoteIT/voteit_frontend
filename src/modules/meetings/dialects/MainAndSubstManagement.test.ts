import { mount } from '@vue/test-utils'
import { expect, test, vi } from 'vitest'
import { createI18n } from 'vue-i18n'

import router from '@/router'
import vuetify from '@/plugins/vuetify'
import MainAndSubstManagement from './MainAndSubstManagement.vue'

vi.mock('@/modules/active/useActive', () => {
  return {
    default() {
      return {
        activeUserIds: { value: [1] }
      }
    }
  }
})

test('Mount component', () => {
  expect(MainAndSubstManagement).toBeTruthy()
  const i18n = createI18n({
    legacy: false,
    messages: { en: {} }
  })
  // @ts-ignore
  const wrapper = mount(MainAndSubstManagement, {
    global: {
      plugins: [i18n, router, vuetify]
    },
    props: {
      group: {
        votes: 2,
        memberships: [
          { user: 1, role: 1 },
          { user: 2, role: 1 }
        ]
      }
    }
  })
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.text()).toContain('1/2')
})
