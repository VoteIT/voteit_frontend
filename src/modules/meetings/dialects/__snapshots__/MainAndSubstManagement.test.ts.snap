// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<span class="text-warning" aria-describedby="v-tooltip-0">1/2</span>
<!--teleport start-->
<!--teleport end-->"
`;
